base_location/models/company.py

```diff
@@ -6,19 +6,19 @@
 
 
 class ResCompany(models.Model):
-
     _inherit = 'res.company'
 
+    better_zip_id = fields.Many2one(
+        'res.better.zip',
+        string='Location',
+        help='Use the city name or the zip code to search the location',
+    )
+
     @api.onchange('better_zip_id')
     def on_change_city(self):
         if self.better_zip_id:
             self.zip = self.better_zip_id.name
+            self.city_id = self.better_zip_id.city_id
```

Comment:
res.company does not have a field 'city_id'

Reply:
You have to rename it to `self.zip_id.city_id`